File: source/WmsDemo/app/src/main/java/com/zero/wmsdemo/MainActivity.java
package com.zero.wmsdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.tv_show)
    TextView tvShow;

    private SuspensionWindowUtil windowUtil;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        windowUtil = new SuspensionWindowUtil(this);
    }

    @OnClick(R.id.btn_show)
    public void onBtnShowClicked() {
        windowUtil.showSuspensionView();
    }

    @OnClick(R.id.btn_hide)
    public void onBtnHideClicked() {
        windowUtil.hideSuspensionView();
    }
}

File: source/jetPackDemo/navigationdemo/src/main/java/com/zero/navigationdemo/MainActivity.kt
package com.zero.navigationdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer
import androidx.navigation.NavController
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var currentNavController: LiveData<NavController>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if(savedInstanceState == null){
            setupBottomNavigationBar()
        }
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        setupBottomNavigationBar()
    }

    private fun setupBottomNavigationBar(){
        val navGraphIds = listOf(R.navigation.shopping,R.navigation.history,R.navigation.mine,R.navigation.about)

        val controller = bottomNavigationView.setupWithNavController1(
            navGraphIds,
            supportFragmentManager,
            R.id.frameLayout,
            intent
        )
        controller.observe(this, Observer {navController ->
            toolbar.setupWithNavController(navController)
        })
        currentNavController = controller
    }

    override fun onSupportNavigateUp(): Boolean {
        return currentNavController?.value?.navigateUp()?:false
    }


}
